src/nft/components/profile/view/ViewMyNftsAsset.tsx

```diff
@@ -0,0 +1,73 @@
+import * as Card from 'nft/components/collection/Card'
+import { AssetMediaType } from 'nft/components/collection/Card'
+import { useBag, useIsMobile, useSellAsset } from 'nft/hooks'
+import { WalletAsset } from 'nft/types'
+import { useMemo, useState } from 'react'
+import { useNavigate } from 'react-router-dom'
+
+const NFT_DETAILS_HREF = (asset: WalletAsset) => `/nfts/asset/${asset.address}/${asset.tokenId}?origin=collection`
+
+export const ViewMyNftsAsset = ({ asset, isSellMode }: { asset: WalletAsset; isSellMode: boolean }) => {
+  const sellAssets = useSellAsset((state) => state.sellAssets)
+  const selectSellAsset = useSellAsset((state) => state.selectSellAsset)
+  const removeSellAsset = useSellAsset((state) => state.removeSellAsset)
+  const cartExpanded = useBag((state) => state.bagExpanded)
+  const toggleCart = useBag((state) => state.toggleBag)
+  const isMobile = useIsMobile()
+  const navigate = useNavigate()
+
+  const [currentTokenPlayingMedia, setCurrentTokenPlayingMedia] = useState<string | undefined>()
```

Comment:
This logic doesn't seem to be working correctly, I can get multiple audio/video nfts to play at once. Testable by impersonating vitalik

Reply:
thanks for catching I see the issue ... need to be setting state in the parent. fixed now